### PR TITLE
Make socketevents mode (poll vs select) configurable via parameter

### PR DESCRIPTION
**Pull request by codablock**
This allows us to test the socket-events mode at startup time instead of at compile time. This PR also adds some randomness in the selected mode for integration testing.

This will later become more useful when epoll support is implemented.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -2307,6 +2307,9 @@ bool AppInitMain()
             connOptions.m_specified_outgoing = connect;
         }
     }
+
+    connOptions.strSocketEventsMode = gArgs.GetArg("-socketevents", "");
```
Comment (UdjinM6) [inline src/init.cpp:2311]: Needs help description

File: src/net.cpp [hunk 2]
```diff
@@ -2695,6 +2710,33 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
...
+        supported.emplace("select");
+        if (!supported.count(strSocketEventsMode)) {
+            std::string strSupported;
+            for (auto& s : supported) {
+                if (!strSupported.empty()) {
+                    strSupported += ", ";
+                }
+                strSupported += s;
+            }
+            clientInterface->ThreadSafeMessageBox(
```
Comment (UdjinM6) [inline src/net.cpp:2733]: should wrap in `if (clientInterface) {`

File: src/net.h [hunk 3]
```diff
@@ -166,6 +166,7 @@ class CConnman
         bool m_use_addrman_outgoing = true;
         std::vector<std::string> m_specified_outgoing;
         std::vector<std::string> m_added_nodes;
+        std::string strSocketEventsMode;
```
Comment (UdjinM6) [inline src/net.h:169]: would prefer enum tbh

File: ci/test_integrationtests.sh [hunk 4]
```diff
@@ -17,8 +17,20 @@ export LD_LIBRARY_PATH=$BUILD_DIR/depends/$HOST/lib
...
+  else
+    SOCKETEVENTS="poll"
+  fi
+fi
+echo "Using socketevents mode: $SOCKETEVENTS"
+EXTRA_ARGS="--dashd-arg=-socketevents=$SOCKETEVENTS"
+
 set +e
-./test/functional/test_runner.py --ci --coverage --failfast --nocleanup --tmpdir=$(pwd)/testdatadirs $PASS_ARGS
+./test/functional/test_runner.py --ci --coverage --failfast --nocleanup --tmpdir=$(pwd)/testdatadirs $PASS_ARGS $EXTRA_ARGS
```
Comment (PastaPastaPasta) [inline ci/test_integrationtests.sh:33]: Soo, does each run of CI randomly choose a different one?

Comment (codablock) [reply ci/test_integrationtests.sh:33]: Yes, every dashd from that one run (including all tests) will use the same setting

File: src/net.cpp (hunk 2 above)
Comment (codablock) [reply src/net.cpp:2733]: Implicitly fixed by removing this code (this logic is in init.cpp now)

File: src/net.h (hunk 3 above)
Comment (codablock) [reply src/net.h:169]: done

File: src/init.cpp (hunk 1 above)
Comment (codablock) [reply src/init.cpp:2311]: done

**Pull request merged**